test/scripts/util.py

```diff
@@ -38,3 +41,56 @@ def print_progress_bar(s: str, current: Union[int, float], total: Union[int, flo
         f"\r{s}: [{'#' * progress}{' ' * (100 - progress)}]" f"{progress}%"
     )
     sys.stdout.flush()
+
+
+def retry_test(max_attempts: int = 5, wait_s: float = 5.0):
+    """Retry pytest for specified number of attempts
+
+    Decorator to retry pytest tests for specific number of attempts.
+    If a test fails, it will retry the the test after waiting
+    a specified number of seconds. If the test fails after retried
+    number of attempts, then it will be reported accordingly.
+
+    Args:
+        max_attempts (int): Number of maximum attempts the test can
+                            be ran.
+        wait_s (int): Time in seconds to wait in between test runs
+                      if a test is to fail.
+    """
+
+    def decorator(test_fun):
+        @wraps(test_fun)
+        def wrapper(*args, **kwargs):
+            retry_count = 1
+
+            while retry_count < max_attempts:
+                try:
+                    return test_fun(*args, **kwargs)
+
+                except AssertionError as assert_error:
+                    assert_message = assert_error.__str__().split("\n")[0]
+                    print(
+                        f'Retry error: "{test_fun.__name__}" '
+                        f"--> {assert_message}."
+                        f"[{retry_count}/{max_attempts - 1}]"
+                        f"Retrying new execution in {wait_s} second(s)"
+                    )
+                    time.sleep(wait_s)
+                    retry_count += 1
+                except serial.SerialException as serial_error:
+                    serial_message = serial_error.__str__().split("\n")[0]
+                    print(
+                        f'Retry error: "{test_fun.__name__}" '
+                        f"--> {serial_message}."
+                        f"[{retry_count}/{max_attempts - 1}]"
+                        f" Retrying new execition in {wait_s} second(s)"
+                    )
+                    time.sleep(wait_s)
+                    retry_count += 1
```

Comment:
Logic looks good. Cool use of a decorator! Since the error handling is identical between the 2 cases can we do `except (AssertionError, serial.SerialException) as err` and avoid duplicating the 9 lines of print-wait-increment code?